govcd/landing_zone_add_on_test.go

```diff
@@ -0,0 +1,126 @@
+
+	slz := createSlz(vcd, check)
+	err := slz.Refresh()
+	check.Assert(err, IsNil)
+
+	addOnCatalog, err := vcd.org.GetCatalogByName(vcd.config.VCD.Catalog.NsxtBackedCatalogName, false)
+	check.Assert(err, IsNil)
+
+	cacheFilePath, err := fetchCacheFile(addOnCatalog, catalogMediaName, check)
+	check.Assert(err, IsNil)
+
+	org, err := vcd.client.GetOrgById(slz.SolutionLandingZoneType.ID)
+	check.Assert(err, IsNil)
+	catalog, err := org.GetCatalogById(slz.SolutionLandingZoneType.Catalogs[0].ID, false)
+	check.Assert(err, IsNil)
+	catItem, err := catalog.GetCatalogItemByName(catalogMediaName, false)
+	check.Assert(err, IsNil)
+
+	createCfg := SolutionAddOnConfig{
+		IsoFilePath:          cacheFilePath,
+		User:                 "administrator",
+		CatalogItemId:        catItem.CatalogItem.ID,
+		AutoTrustCertificate: true,
+	}
+	solutionAddOn, err := vcd.client.CreateSolutionAddOn(createCfg)
+	check.Assert(err, IsNil)
+	PrependToCleanupListOpenApi(solutionAddOn.DefinedEntity.DefinedEntity.ID, check.TestName(), types.OpenApiPathVersion1_0_0+types.OpenApiEndpointRdeEntities+solutionAddOn.DefinedEntity.DefinedEntity.ID)
+
+	// Get all
+	allSolutionAddOns, err := vcd.client.GetAllSolutionAddons(nil)
+	check.Assert(err, IsNil)
+	check.Assert(len(allSolutionAddOns) >= 1, Equals, true) // VCD has a few baked in Solution Add-Ons (e.g. 'service-account-solutions-system-user', 'vmware.solution-addon-landing-zone-1.0.0')
+
+	foundId := false
+	for addOnIndex := range allSolutionAddOns {
+		if allSolutionAddOns[addOnIndex].RdeId() == solutionAddOn.RdeId() {
+			foundId = true
+			break
+		}
+	}
+	check.Assert(foundId, Equals, true)
+
+	// Get all with filter
+	queryParams := queryParameterFilterAnd("id=="+solutionAddOn.RdeId(), nil)
+	filteredSolutionAddon, err := vcd.client.GetAllSolutionAddons(queryParams)
+	check.Assert(err, IsNil)
+	check.Assert(len(filteredSolutionAddon), Equals, 1)
+
+	// By ID
+	sao, err := vcd.client.GetSolutionAddonById(solutionAddOn.RdeId())
+	check.Assert(err, IsNil)
+	check.Assert(sao.RdeId(), Equals, solutionAddOn.RdeId())
+
+	// By Name
+	saoByName, err := vcd.client.GetSolutionAddonByName(solutionAddOn.DefinedEntity.DefinedEntity.Name)
+	check.Assert(err, IsNil)
+	check.Assert(saoByName.RdeId(), Equals, solutionAddOn.RdeId())
+
+	// Delete
+	err = sao.Delete()
+	check.Assert(err, IsNil)
+
+	// Verify no more Add-Ons remaining
+	allSolutionAddOnsAfterCleanup, err := vcd.client.GetAllSolutionAddons(nil)
+	check.Assert(err, IsNil)
+	check.Assert(len(allSolutionAddOnsAfterCleanup), Equals, len(allSolutionAddOns)-1)
+}
+
+func fetchCacheFile(catalog *Catalog, fileName string, check *C) (string, error) {
+	pwd, err := os.Getwd()
+	check.Assert(err, IsNil)
+	cacheDirPath := pwd + "/test-resources/cache"
+	cacheFilePath := cacheDirPath + "/" + fileName
+	printVerbose("# Using '%s' file to cache Solution Add-On\n", cacheFilePath)
+
+	if _, err := os.Stat(cacheFilePath); errors.Is(err, os.ErrNotExist) {
```

Comment:
This check will fail if `test-resources/cache` exists, but it is not a directory.

Reply:
Now it checks if it is a file, removes it and creates a directory instead.